zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UsageMetricHandler.java

```diff
@@ -47,58 +55,119 @@ public boolean handlesRecord(final Record<UsageMetricRecordValue> record) {
 
   @Override
   public List<String> generateIds(final Record<UsageMetricRecordValue> record) {
-    final long key = record.getKey();
-    return record.getValue().getCounterValues().keySet().stream()
-        .map(tenantId -> ID_PATTERN.formatted(key, tenantId))
-        .toList();
+    return Collections.singletonList(String.valueOf(record.getKey()));
   }
 
   @Override
-  public UsageMetricsEntity createNewEntity(final String id) {
-    return new UsageMetricsEntity().setId(id);
+  public UsageMetricsBatch createNewEntity(final String id) {
+    return new UsageMetricsBatch(id, new ArrayList<>(), new ArrayList<>());
   }
 
   @Override
   public void updateEntity(
-      final Record<UsageMetricRecordValue> record, final UsageMetricsEntity entity) {
-    final var recordValue = record.getValue();
-    final var tenantId = parseTenantId(entity.getId());
+      final Record<UsageMetricRecordValue> record, final UsageMetricsBatch usageMetricsBatch) {
 
+    final var recordValue = record.getValue();
     final var eventType = mapEventType(recordValue.getEventType());
 
-    if (eventType != null) {
-      final var tenantValue = recordValue.getCounterValues().get(tenantId);
-      entity
-          .setEventTime(DateUtil.toOffsetDateTime(recordValue.getStartTime()))
-          .setEventType(eventType)
-          .setEventValue(tenantValue)
-          .setTenantId(tenantId)
-          .setPartitionId(record.getPartitionId());
-    } else {
+    if (eventType == null) {
       LOGGER.warn("Unsupported event type: {}", recordValue.getEventType());
+      return;
     }
+
+    final var recordKey = record.getKey();
+    final var partitionId = record.getPartitionId();
+    final var timestamp = DateUtil.toOffsetDateTime(record.getTimestamp());
+    final var collection = usageMetricsBatch.variables();
+    final var tuCollection = usageMetricsBatch.tuVariables();
+
+    recordValue
+        .getCounterValues()
+        .forEach(
+            (tenantId, counter) ->
+                collection.add(
+                    createUsageMetricEntity(
+                        recordKey, partitionId, timestamp, tenantId, eventType, counter)));
+
+    recordValue
+        .getSetValues()
+        .forEach(
+            (tenantId, set) -> {
+              collection.add(
+                  createUsageMetricEntity(
+                      recordKey, partitionId, timestamp, tenantId, eventType, set.size()));
```

Comment:
🔧 I'd still not save TU counts. Customers might see it and think it can be used as TU or they will get confused and use it for something else. And once it's live we can't remove it easily as it will be a breaking change.

Reply:
I'd simply not save TU in the regular table/index. TU is only saved in its own table/index.
Yes, let's change it for RDBMS, too then. They should behave the same.